pkgs/build-support/libredirect/default.nix

```diff
@@ -15,8 +15,17 @@ stdenv.mkDerivation rec {
 
   buildPhase = ''
     runHook preBuild
-
-    $CC -Wall -std=c99 -O3 -fPIC -ldl -shared \
+    ${if stdenv.isDarwin then ''
```

Comment:
```suggestion
    ${if stdenv.hostPlatform.system == "aarch64-darwin" then ''
```

Reply:
I guess we could do this, if we wanted to start treating x86_64 and aarch64 differently, my hope was for as little custom logic as possible. If we can build arm64e with nixpkgs somehow, we might get away without using the system clang.

Reply:
If this is not done, the PR as it is will break the Hydra build on both platforms.